src/Twig/PaginatorRuntime.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace SumoCoders\FrameworkCoreBundle\Twig;
+
+use SumoCoders\FrameworkCoreBundle\Pagination\Paginator;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Twig\Environment;
+use Twig\Extension\RuntimeExtensionInterface;
+
+class PaginatorRuntime implements RuntimeExtensionInterface
+{
+    private Environment $twig;
+    private RequestStack $requestStack;
+
+    public function __construct(Environment $twig, RequestStack $requestStack)
+    {
+        $this->twig = $twig;
+        $this->requestStack = $requestStack;
+    }
+
+    public function renderPagination(Paginator $paginator): string
+    {
+        $request = $this->getRequest();
+
+        if (null !== $this->requestStack->getParentRequest()) {
+            throw new \RuntimeException('The request aware route generator can not guess the route when used in a sub-request, pass the "routeName" option to use this generator.');
```

Comment:
Is deze lijn niet te lang?

Reply:
Error was ook niet volledig relevant voor onze use case, dus aangepast naar kortere boodschap in 4aed958